.circleci/config.yml

```diff
@@ -2,13 +2,73 @@
 version: 2.0
 jobs:
   build:
+    working_directory: /usr/src/artman
     docker:
-      - image: python:3.6
+      - image: docker:17.05.0-ce-git
     steps:
+      - checkout
+      - setup_remote_docker
       - run:
-          name: Hello world!
-          command: echo "Hello world!"
-    working_directory: /usr/src/artman/
+          name: Finger crossed!
+          command: |
+            echo "Finger crossed!"
+      - restore_cache:
+          keys:
+            - artman-{{ .Branch }}
+          paths:
+            - /caches/artman.tar
+      - run:
+          name: Load Artman Docker image layer cache
+          command: |
+            set +o pipefail
+            docker load -i /caches/artman.tar | true
+      - run:
+          name: Build application Docker image
+          command: |
+            echo "ADD . /artman" >> Dockerfile
+            echo "RUN pip uninstall -y googleapis-artman" >> Dockerfile
+            echo "RUN pip install -e /artman" >> Dockerfile
+            docker build --cache-from=artman -t artman .
+      - run:
+          name: Save Artman Docker image layer cache
+          command: |
+            mkdir -p /caches
+            docker save -o /caches/artman.tar artman
+      - save_cache:
+          key: artman-{{ .Branch }}
+          paths:
+            - /caches/artman.
+      - run:
+          name: Run golden tests
+          command: |
+            docker run -it -e RUNNING_IN_ARTMAN_DOCKER=True -v /tmp/reports:/tmp/reports artman /bin/bash -c "pip install pytest; git clone https://github.com/googleapis/googleapis /artman/test/golden/googleapis; py.test /artman/test/golden/artman_golden_test.py --googleapis-dir=/artman/test/golden/googleapis"
+      # When build performance is a concern, consider separting this job into
+      # a workflow with multiple jobs and have the smoketest run in parrall
+      # with additional paied containers.
+      - run:
+          name: Run smoketests
+          command: |
+            mkdir /tmp/reports
+            echo "this is a test" >> /tmp/reports/smoketest.log
```

Comment:
> `>> /tmp/reports/smoketest.log`

Is it expected that smoketest.log already exists? If so, we should be more specific, for example, "Artman smoketest at {commit_id}, {time}". If not, no need to append to the log file.

Reply:
Removed this line. I prefer to use constant file name for two reasons: 1) each execution will uses a new container, so it is not going to cause conflicts. 2) circleci 2.0 currently has a limited test analysis/reporting system, and I plan to create a GAE-based job which will trigger cadence test via cron task, and collect test result by checking circleci artifacts. Using fixed file name will make the artifacts fetching a bit easy.